### PR TITLE
Create main.yml

### PR DESCRIPTION
**Pull request by IlyaMoskva**
Issue #7 : Workflow for build and test

### CODE REVIEW

**Comment by IlyaMoskva** [comment]
Blocked by PR #10

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+val compileKotlin: KotlinCompile by tasks
+val compileTestKotlin: KotlinCompile by tasks
+compileKotlin.run {
+    kotlinOptions.jvmTarget = "1.8"
+}
+compileTestKotlin.run {
+    kotlinOptions.jvmTarget = "1.8"
+}
```
Comment (g4s8) [inline .github/workflows/main.yml:28]: @IlyaMoskva do we really need to use `1.8` JVM?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+dependencies {
+    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion")
+    implementation("org.jetbrains.kotlinx:kotlinx-cli:0.3")
+
+    testImplementation("org.testng:testng-6.14.3")
+}
+
+tasks.test {
+    // enable TestNG support (default is JUnit)
+    useTestNG()
```
Comment (g4s8) [inline .github/workflows/main.yml:39]: @IlyaMoskva why testNG, not Junit?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    repositories {
+        mavenLocal()
+        mavenCentral()
+    }
+}
+dependencies {
+    ktlint("com.pinterest:ktlint:0.37.1") {
+        exclude("com.pinterest.ktlint", "ktlint-ruleset-standard")
+    }
+    ktlint("org.cqfn.diktat:diktat-rules:0.1.2")
```
Comment (g4s8) [inline .github/workflows/main.yml:58]: @IlyaMoskva I don't think we need these dependencies

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+}
+task<JavaExec>("diktat") {
+    setGroup("verification")
+    setDescription("Check Kotlin code style.")
+    mainClass.set("com.pinterest.ktlint.Main")
+    classpath = ktlint
+    args("--debug", "src/**/*.kt")
+}
```
Comment (g4s8) [inline .github/workflows/main.yml:66]: @IlyaMoskva this gradle task seems to be unrelated

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    testImplementation("org.testng:testng-6.14.3")
+}
+
+tasks.test {
+    // enable TestNG support (default is JUnit)
+    useTestNG()
+}
+
+application {
+    mainClass.set("com.degitx.simulator.Simulator")
```
Comment (g4s8) [inline .github/workflows/main.yml:43]: @IlyaMoskva it seems we need a main class stub to test this gradle script. Also, please check my comment about package name

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+group = "com.degitx.simulator"
```
Comment (g4s8) [inline .github/workflows/main.yml:3]: @IlyaMoskva we're using `org.cqfn.*` as a base for group id. I think we can use `org.cqfn.degitx.simulator` as a package name here

**Review by g4s8 — CHANGES_REQUESTED** [review]
@IlyaMoskva please check my comments

File: .github/workflows/main.yml (hunk 2 above)
Comment (IlyaMoskva) [reply .github/workflows/main.yml:39]: @g4s8 Why not TestNG? They are very similar, so choice based on this comparison: https://www.guru99.com/junit-vs-testng.html.

File: .github/workflows/main.yml (hunk 6 above)
Comment (IlyaMoskva) [reply .github/workflows/main.yml:3]: @g4s8 Thanks for clarification, I'll use it.

File: .github/workflows/main.yml (hunk 2 above)
Comment (g4s8) [reply .github/workflows/main.yml:39]: @IlyaMoskva ok, let's use testNG if you think it's better here

**Review by g4s8 — APPROVED** [review]
@IlyaMoskva thanks
